sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -543,34 +573,47 @@ def _fill_predictor_node_array(predictor_nodes, grower_node,
     if grower_node.is_leaf:
         # Leaf node
         node['is_leaf'] = True
-        return next_free_idx + 1
+        return next_free_node_idx + 1, next_free_bitset_idx
```

Comment:
Note for reviewers: diff looks bigger than it is because we indented everything to the left